be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -1186,6 +1187,13 @@ void SegmentIterator::_vec_init_lazy_materialization() {
             _is_pred_column[cid] = true;
             pred_column_ids.insert(cid);
 
+            if (predicate->need_to_clone()) {
```

Comment:
Not modify here, modify it in 
Status SegmentIterator::init(const StorageReadOptions& opts) {
    _opts = opts;
    if (!opts.column_predicates.empty()) {
        _col_predicates = opts.column_predicates;
    }

during init process, I think we could clone predicates and replace it.